go-runtime/compile/build.go

```diff
@@ -50,6 +50,12 @@ type mainModuleContext struct {
 	Replacements []*modfile.Replace
 }
 
+type ModifyFilesTransaction interface {
```

Comment:
additional interface defined here so that `compile` package doesn't create an import loop with`buildengine`